test/e2e/alphaupdate/update_test.go

```diff
@@ -105,6 +106,51 @@ var _ = Describe("kubebuilder", func() {
 			By("validating custom code preservation")
 			validateCustomCodePreservation(mockProjectDir)
 		})
+
+		It("should abort on merge conflicts when --force is not used", func() {
+			By("creating mock project with kubebuilder v4.5.2")
+			createMockProject(mockProjectDir, binFromVersionPath)
+
+			By("injecting conflicting code that will cause merge conflicts")
+			injectConflictingCode(mockProjectDir)
+
+			By("initializing git repository and committing mock project")
+			initializeGitRepo(mockProjectDir)
+
+			By("running alpha update without --force flag")
+			cmd := exec.Command(kbc.BinaryName, "alpha", "update",
+				"--from-version", fromVersion, "--to-version", toVersion, "--from-branch", "main")
+			cmd.Dir = mockProjectDir
+			output, err := cmd.CombinedOutput()
+
+			By("expecting the command to fail due to merge conflicts")
+			Expect(err).To(HaveOccurred())
+			Expect(string(output)).To(ContainSubstring("Merge conflicts detected"))
+			Expect(string(output)).To(ContainSubstring("merge aborted due to conflicts"))
+		})
+
+		It("should commit with conflict markers when --force is used", func() {
+			By("creating mock project with kubebuilder v4.5.2")
```

Comment:
Could we not generate exactly the same mock with 4.5.0 ?
It will generate a mock with conflicts